next/src/modules/dashboardPage/pages/DashboardPage.tsx

```diff
@@ -41,11 +42,15 @@ const DashboardPage = () => {
   const { status } = useSession();
   const loading = status === 'loading';
 
+  const queryClient = useQueryClient();
+
+  const getCachedFragment = (id: string) =>
+    queryClient.getQueryData<EcgFragment>(['record', id]);
```

Comment:
I'm trying to learn more about react and next, so can you give me some more context about how this caching works? Just a comment here is enough

Reply:
We are using [ReactQuery](https://tanstack.com/query/latest/docs/react/overview) - a fantastic library for handling some of our requests.
It provides an excellent caching and cache revalidation mechanism.
The basic principle is very simple. We pass a key and a request function - after the request is completed, the cache will be available using the key provided earlier.
In this specific case, we are retrieving a cached fragment that was fetched in another component (chart).